swap_meet/item.py

```diff
@@ -1,2 +1,37 @@
+import uuid
+
 class Item:
-    pass
\ No newline at end of file
+    def __init__(self, id = None, condition = 0):
+        if id is  None:
+            self.id =  uuid.uuid4().int 
+        else:
+            self.id = id 
+        
+        if isinstance(condition, (int, float)):
+            if 0 <= condition <= 5:
+                self.condition = condition 
+            else:
+                raise ValueError("Condition must be between 0 and 5")
+        else: 
+            raise TypeError("Condition must be a number")
+
+    
+    def get_category(self):
+        return self.__class__.__name__
+    
+    def __str__(self):
+        return f"An object of type Item with id {self.id}."
```

Comment:
This is the thing I mentioned in the above subclasses - if you used the `get_category` function in the `__str__` override, then you could use the call to `super().__str__()` in the subclasses! e.g.
```python
# in Item class
def __str__(self):
  return f"An object of type {self.get_category()} with id {self.id}."

# in subclass that inherits from Item

def __str__(self):
      return f"An object of type {super().__str__()} with id {self.id}."

# or alternatively in electronics subclass for instance:
    def __str__(self):
        return f"An object of type {self.get_category()} with id {self.id}. It takes up a {self.width} by {self.length} sized space."
```